presence_channel.go

```diff
@@ -0,0 +1,332 @@
+package pusher
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+var (
+	// ErrNotSubscribed is returned by functions that require the channel to be
+	// subscribed before being called.
+	ErrNotSubscribed = fmt.Errorf("not subscribed")
+	// ErrMissingMe indicates a presence channel was subscribed to, but the pusher
+	// server violated the message protocol by not providing a member for the
+	// current user.
+	ErrMissingMe = fmt.Errorf("missing member for current user")
+)
+
+// Member represents a channel member.
+type Member struct {
+	ID string
+	// Info is the JSON set by the auth server. Do not modify the underlying byte
+	// slice, it's shared by all instances of the member.
+	Info json.RawMessage
+}
+
+// PresenceChannel provides information about the users that are currently
+// subscribed.
+//
+// Note: Bind() does not fire pusher:member_added/removed, use
+// BindMemberAdded/Removed() instead.
+type PresenceChannel interface {
+	Channel
+
+	// BindMemberAdded returns a channel that receives a Member value when a user
```

Comment:
nit: maybe we should add a note to the `Bind*` method comments saying that the corresponding `Unbind*` method must be called when done reading from the channel so the requirement is explicit